src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -177,12 +187,48 @@ async def _run_impl(self) -> None:
                 else:
                     self._buffer.update(sample)
 
+                # count the number of samples and send out a trigger when it matches
+                # the number of samples to wait for.
+                received_samples_count += 1
+                if self._wait_for_num_samples != 0:
+                    if received_samples_count == self._wait_for_num_samples:
+                        received_samples_count = 0
+                        await wait_for_samples_sender.send(None)
+
         except asyncio.CancelledError:
             _logger.info("MovingWindow task has been cancelled.")
             raise
 
         _logger.error("Channel has been closed")
 
+    def set_sample_counter(self, num_samples: int) -> None:
+        """Set the number of samples to wait for until the sample counter triggers.
+
+        Args:
+            num_samples: The number of samples to wait for.
+
+        Raises:
+            ValueError: if the number of samples is less than or equal to zero.
+        """
+        if num_samples <= 0:
+            raise ValueError(
+                "The number of samples to wait for should be greater than zero."
+            )
+        self._wait_for_num_samples = num_samples
+
+    def new_sample_count_receiver(self) -> Receiver[None]:
```

Comment:
If the above suggestion is applied, I would rename this for consistency:

```suggestion
    def new_wait_for_samples_event_receiver(self) -> Receiver[None]:
```

We could go with something different than `wait_for_samples_event` as it is a bit long, but the 4 methods should use the same term IMHO.